ghost/collections/src/Collection.ts

```diff
@@ -22,6 +23,8 @@ export class Collection {
     updatedAt: Date;
     deleted: boolean;
 
+    posts: PostDTO[];
```

Comment:
I'm wondering if this is the best design type here. By requiring a `tile` and `slug` property, we need to always fetch the related data in the DB when using a SQL backed repository. 

That could well be fine if we only want to expose the title/slug from the API - but if we want to include whole post objects then we'd want to either:
1. Do another fetch of the related data to get all of it in the controller/usecase layer
2. Expand this `PostDTO` type to include all the fields which we do wanna use in the API

I think if the fieldset is small then it's okay - but if we think we're gonna wanna expose the entire Post object then we'll be bloating the Collection entity.

So TL;DR - if we just going to respond with a subset of the post fields I think this makes sense, otherwise I would do something like:

```ts
class Collection {
  // stuff...
  posts: ObjectID[]

  addPost(post: PostDTO): void {
    if (this.posts.includes(post.id)) {
      return;
    }
    if (this.type === 'automatic') {
      if (!nql(this.filter).queryJSON(post)) {
        throw new Error('Post doesn't match filter');
      }
    }
    this.posts.push(post.id);
  }
  // stuff..
}
```

This gives us the benefit of the fuller Post object for validation but keeps the exposed fields small to allow us to make fewer DB queries.

Reply:
Agreed on minimizing the the size of exposed fields 👌  

For now we don't have a concrete spec for which fields we'd want to keep around and which ones we don't. My thinking is to start from this basic structure that would be easy to filter right away with NQL and than optimize/reshuffle things as we get feedback. 

An extra thing we'd want to start keeping track straight away is order or items in the collection. So I think if we go with shrinking of the "posts" relation we'd go to something like:
```
{
  id: ObjectID, // to be able to modify the relational object individually
  post_id: ObjectID, 
  sort_order: number
}
```